addons/mail/static/src/components/message_list/message_list.js

```diff
@@ -371,29 +363,6 @@ class MessageList extends Component {
         this.threadViewer.markComponentHintProcessed(hint);
     }
 
-    /**
-     * @private
-     */
-    _checkThreadMarkAsRead() {
-        const thread = this.threadViewer.thread;
-        const threadCache = this.threadViewer.threadCache;
-        if (!threadCache) {
-            return;
-        }
-        if (threadCache.messages.length === 0) {
-            return;
-        }
-        if (!this.mostRecentMessageRef) {
-            return;
-        }
-        if (
-            threadCache === thread.mainCache &&
-            this.mostRecentMessageRef.isPartiallyVisible()
-        ) {
-            thread.markAsSeen();
-        }
```

Comment:
This is the logic I was talking about, it must be "kept" in model for all flux that intend to set mark as read, the appropriate conditions must still be met.

Reply:
Probably need to keep some form of _update/scroll handlers that set some attributes on the models, so that the model can check them to decide if mark as seen should be sent.

Basically state of what is "viewed" must be forwarded from components to models.